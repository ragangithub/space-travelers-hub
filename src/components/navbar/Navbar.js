import React from 'react';
import { NavLink } from 'react-router-dom';

import logo from '../../assets/logo.png';

const Navbar = () => (
  <>
    <nav className="navBar">
      <div className="logo_title">
        <NavLink to="/" exact>
          <img src={logo} alt="logo" className="logo" />
        </NavLink>
        <a href="#home" className="title">
          Space Travelers&apos; Hub
        </a>
      </div>
      <div className="navItems">
        <ul className="navLinks">
          <NavLink to="/" exact className="menu_item">
            Rockets
          </NavLink>
          <div className="border" />
          <NavLink to="/missions" exact className="menu_item">
            Missions
          </NavLink>

          <div className="border" />
          <NavLink to="/profiles" className="menu_item">
            My Profile
          </NavLink>
        </ul>
      </div>
    </nav>
  </>
);

export default Navbar;
