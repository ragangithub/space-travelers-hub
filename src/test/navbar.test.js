import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/navbar/Navbar';

describe('Navbar component', () => {
  test('renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
