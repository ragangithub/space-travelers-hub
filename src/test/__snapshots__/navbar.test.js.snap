// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test navbar render 1`] = `
<nav
  className="navBar"
>
  <div
    className="logo_title"
  >
    <a
      aria-current="page"
      className="active"
      exact={true}
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo"
        src="logo.png"
      />
    </a>
    <a
      className="title"
      href="#home"
    >
      Space Travelers' Hub
    </a>
  </div>
  <div
    className="navItems"
  >
    <ul
      className="navLinks"
    >
      <a
        aria-current="page"
        className="menu_item active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
      <a
        className="menu_item"
        href="/mission"
        onClick={[Function]}
      >
        Missions
      </a>
      <div
        className="border"
      />
      <a
        className="menu_item"
        href="/profiles"
        onClick={[Function]}
      >
        My Profile
      </a>
    </ul>
  </div>
</nav>
`;
